contrib/src/main/java/com/datatorrent/contrib/cassandra/AbstractUpsertOutputOperator.java

```diff
@@ -0,0 +1,1401 @@
+            getter = PojoUtils.createGetter(tuplePayloadClass, getterExpr, Object.class);
+            break;
+        }
+        String resolvedColumnName = colNamesMap.get(aFieldName.toLowerCase());
+        if (overridingColnamesMap.containsKey(aFieldName)) {
+          resolvedColumnName = overridingColnamesMap.get(aFieldName);
+        }
+        getters.put(resolvedColumnName, getter);
+      }
+    }
+  }
+
+  private void registerCodecs()
+  {
+    complexTypeCodecs = getCodecsForUserDefinedTypes();
+    if (complexTypeCodecs != null) {
+      CodecRegistry registry = cluster.getConfiguration().getCodecRegistry();
+      if (cluster.getConfiguration().getProtocolOptions().getProtocolVersion().toInt() < 4) {
+        LOG.error("Custom codecs are not supported for protocol version < 4");
+        throw new RuntimeException("Custom codecs are not supported for protocol version < 4");
+      }
+      for (String typeCodecStr : complexTypeCodecs.keySet()) {
+        TypeCodec codec = complexTypeCodecs.get(typeCodecStr);
+        registry.register(codec);
+        userDefinedTypesClass.put(typeCodecStr, codec.getJavaType().getRawType());
+      }
+    } else {
+      complexTypeCodecs = new HashMap<>();
+    }
+  }
+
+  private void registerNonPKColumnDefinitions(final TableMetadata tableMetadata)
+  {
+    List<ColumnMetadata> colInfoForTable = tableMetadata.getColumns();
+    for (ColumnMetadata aColumnDefinition : colInfoForTable) {
+      if (aColumnDefinition.getType().isCollection()) {
+        collectionColumns.add(aColumnDefinition.getName());
+      }
+      if (!pkColumnNames.contains(aColumnDefinition.getName())) {
+        columnDefinitions.put(aColumnDefinition.getName(), aColumnDefinition.getType());
+        regularColumns.add(aColumnDefinition.getName());
+      }
+      parseForSpecialDataType(aColumnDefinition);
+    }
+  }
+
+  private void parseForSpecialDataType(final ColumnMetadata aColumnDefinition)
+  {
+    switch (aColumnDefinition.getType().getName()) {
+      case COUNTER:
+        counterColumns.add(aColumnDefinition.getName());
+        break;
+      case MAP:
+        mapColumns.add(aColumnDefinition.getName());
+        break;
+      case SET:
+        setColumns.add(aColumnDefinition.getName());
+        break;
+      case LIST:
+        listColumns.add(aColumnDefinition.getName());
+        break;
+      case UDT:
+        userDefinedTypeColumns.add(aColumnDefinition.getName());
+        break;
+      default:
+        break;
+    }
+  }
+
+  private void registerPrimaryKeyColumnDefinitions(final TableMetadata tableMetadata)
+  {
+    List<ColumnMetadata> primaryKeyColumns = tableMetadata.getPrimaryKey();
+    for (ColumnMetadata primaryColumn : primaryKeyColumns) {
+      columnDefinitions.put(primaryColumn.getName(), primaryColumn.getType());
+      pkColumnNames.add(primaryColumn.getName());
+      parseForSpecialDataType(primaryColumn);
+    }
+  }
+
+  private void generatePreparedStatements()
```

Comment:
There is an object mapper equivalent from datastax itself but there are aspects like 

1. If exists clauses
2. TTLs 
3. Consistency Levels 

are typically not part of the DSL language and there are other messy issues of : 

1. Not all of the above constructs are occurring in a sequence of strings after the query is formed in a DSL generated statement and hence cant simply append etc. 
2. Not all of them give control of list placement styles etc. 

Hence had to handcraft the permutations of all prepared statement strings rather than using any DSL.